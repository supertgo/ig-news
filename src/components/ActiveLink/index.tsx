import Link, { LinkProps } from 'next/link';
import { useRouter } from 'next/router';
import { cloneElement } from 'react';

export type ActiveLinkProps = {
  children: React.ReactElement;
} & LinkProps;

const ActiveLink = ({ children, ...props }: ActiveLinkProps) => {
  const { asPath } = useRouter();
  const isActive = asPath === props.href;

  return <Link {...props}>{cloneElement(children, { isActive })}</Link>;
};

export default ActiveLink;
