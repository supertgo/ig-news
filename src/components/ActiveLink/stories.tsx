import { Story, Meta } from '@storybook/react';
import ActiveLink from '.';

export default {
  title: 'ActiveLink',
  component: ActiveLink
} as Meta;

export const Default: Story = () => <ActiveLink />;
