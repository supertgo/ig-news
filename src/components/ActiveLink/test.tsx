import { render, screen } from '@testing-library/react';

import ActiveLink from '.';

jest.mock('next/router', () => ({
  useRouter() {
    return {
      route: '/',
      pathname: '',
      query: '',
      asPath: ''
    };
  }
}));

describe('<ActiveLink />', () => {
  it('should render the heading', () => {
    const { container } = render(
      <ActiveLink href="/link">
        <a>link</a>
      </ActiveLink>
    );

    expect(screen.getByRole('link', { name: /link/i })).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
