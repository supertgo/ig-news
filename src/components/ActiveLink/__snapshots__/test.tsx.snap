// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActiveLink /> should render the heading 1`] = `
<a
  href="/link"
>
  link
</a>
`;
